pkg/util/util_test.go

```diff
@@ -0,0 +1,233 @@
+	. "github.com/rancher/submariner/pkg/util"
+)
+
+var _ = Describe("Util", func() {
+	Describe("Function ParseSecure", func() {
+		Context("With a valid token input", func() {
+			It("should return a valid Secure object", func() {
+				apiKey := "AValidAPIKeyWithLengthThirtyTwo!"
+				secretKey := "AValidSecretKeyWithLength32Chars"
+				secure, err := ParseSecure(apiKey + secretKey)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(secure.ApiKey).To(Equal(apiKey))
+				Expect(secure.SecretKey).To(Equal(secretKey))
+			})
+		})
+
+		Context("With an invalid token input", func() {
+			It("should return an error", func() {
+				_, err := ParseSecure("InvalidToken")
+				Expect(err).To(HaveOccurred())
+			})
+		})
+	})
+	
+	Describe("Function GetLocalIP", func() {
+		It("should return a non-nil IP", func() {
+			Expect(GetLocalIP()).NotTo(BeNil())
+		})
+	})
+
+	Describe("Function FlattenColors", func() {
+		Context("With a single element", func() {
+			It("should return the element", func() {
+				Expect(FlattenColors([]string{"blue"})).To(Equal("blue"))
+			})
+		})
+
+		Context("With multiple elements", func() {
+			It("should return a comma-separated string containing all the elements", func() {
+				Expect(FlattenColors([]string{"red", "white", "blue"})).To(Equal("red,white,blue"))
+			})
+		})
+		
+		Context("With no elements", func() {
+			It("should return an empty string", func() {
+				Expect(FlattenColors([]string{})).To(Equal(""))
+			})
+		})
+	})
+
+	Describe("Function GetLocalCluster", func() {
+		It("should return a valid SubmarinerCluster object", func() {
+			clusterId := "east"
+			clusterCidr := []string{"1.2.3.4/16"}
+			serviceCidr := []string{"5.6.7.8/16"}
+			colorCodes := []string{"red"}
+			cluster, err := GetLocalCluster(types.SubmarinerSpecification {
+				ClusterId: clusterId,
+				ClusterCidr: clusterCidr,
+				ServiceCidr: serviceCidr,
+				ColorCodes: colorCodes,
+			})
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cluster.ID).To(Equal(clusterId))
+			Expect(cluster.Spec.ClusterID).To(Equal(clusterId))
+			Expect(cluster.Spec.ServiceCIDR).To(Equal(serviceCidr))
+			Expect(cluster.Spec.ClusterCIDR).To(Equal(clusterCidr))
+			Expect(cluster.Spec.ColorCodes).To(Equal(colorCodes))
+		})
+	})
+
+	Describe("Function GetLocalEndpoint", func() {
+		It("should return a valid SubmarinerEndpoint object", func() {
+			subnets := []string{"1.2.3.4/16"}
+			endpoint, err := GetLocalEndpoint("east", "backend", map[string]string{}, false, subnets)
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(endpoint.Spec.ClusterID).To(Equal("east"))
+			Expect(endpoint.Spec.CableName).To(HavePrefix("submariner-cable-east-"))
+			Expect(endpoint.Spec.Hostname).NotTo(Equal(""))
+			Expect(endpoint.Spec.PrivateIP).NotTo(BeNil())
+			Expect(endpoint.Spec.Backend).To(Equal("backend"))
+			Expect(endpoint.Spec.Subnets).To(Equal(subnets))
+			Expect(endpoint.Spec.NATEnabled).To(Equal(false))
+		})
+	})
```

Comment:
Testing for Nil return value??

Reply:
You mean an explicit check like 

    Expect(endpoint).ToNot(BeNil())

? We can but that's indirectly covered by dereferencing endpoint, which would panic and fail if nil.